ratis-common/src/main/java/org/apache/ratis/protocol/RaftPeer.java

```diff
@@ -132,6 +170,8 @@ private RaftPeerProto buildRaftPeerProto() {
         .setId(getId().toByteString());
     Optional.ofNullable(getAddress()).ifPresent(builder::setAddress);
     Optional.ofNullable(getDataStreamAddress()).ifPresent(builder::setDataStreamAddress);
+    Optional.ofNullable(getClientAddress()).ifPresent(builder::setClientAddress);
+    Optional.ofNullable(getAdminAddress()).ifPresent(builder::setAdminAddress);
```

Comment:
Is there a need in build() to check whether these address are different or it is ok to reuse the same address for these fields (I am not sure, just asking)

Reply:
I guess this is answered before: client and admin can reuse address.

Reply:
Yes, they share address by default for backward compatibility.